lib/spack/spack/spec.py

```diff
@@ -4524,6 +4524,21 @@ def do_parse(self):
         except spack.parse.ParseError as e:
             raise SpecParseError(e)
 
+        # Generate lookups for git-commit-based versions
+        for spec in specs:
+            # Cannot do lookups for versions in anonymous specs
+            # Only allow concrete versions using git for now
+            if spec.name and spec.versions.concrete and spec.version.is_commit:
+                pkg = spec.package
+                if hasattr(pkg, 'git'):
+                    import spack.fetch_strategy as fetch
+                    spec.version.generate_commit_lookup(
```

Comment:
Just testing this locally - for large packages there will be a long pause that feels like a freeze for spack install package@commit. Maybe a log info here would be appropriate?
```suggestion
                    tty.info(f"Generating commit lookup for {pkg.name}...")
                    spec.version.generate_commit_lookup(
```